css/datagrid/datagrid.css

```diff
@@ -116,25 +116,41 @@
 
 .highcharts-datagrid-table thead tr {
     height: 32px;
+    border-top: 1px solid var(--highcharts-neutral-color-20);
+}
+
+.highcharts-datagrid-table thead tr:first-child,
+.highcharts-datagrid-table thead th.highcharts-datagrid-column-first {
+    border-top: none;
 }
 
 .highcharts-datagrid-table thead th {
     position: relative;
     border-left: 1px solid var(--highcharts-neutral-color-20);
 }
 
-.highcharts-datagrid-head-cell-content {
-    display: block;
+.highcharts-datagrid-header-cell-content {
+    display: flex;
+    position: absolute;
     overflow: hidden;
     text-overflow: ellipsis;
     padding-right: 10px;
     padding-left: 3px;
     width: 100%;
+    height: 100%;
+    top: 0;
+    align-items: center;
     -webkit-user-select: none; /* Safari */
     -ms-user-select: none; /* IE 10 and IE 11 */
     user-select: none; /* Standard syntax */
 }
 
+.highcharts-datagrid-header-cell-content span {
+    flex: 1;
+    overflow: hidden;
+    text-overflow: ellipsis;
```

Comment:
I suggest adding `white-space: nowrap;` (or some equivalent) to avoid linebreaks and double elipsis in our default style. Devs can override if needed.

<img width="108" alt="Skjermbilde 2024-08-23 kl  08 56 48" src="https://github.com/user-attachments/assets/fddc26be-4592-48d1-b8c3-20d0b4e72a32">

Reply:
I'm not convinced, let's consider a scenario.
1. The `th` has rowSpan so the text is displayed in a few lines.
2. In the case of `white-space: nowrap;` it will be displayed in a single line with dots. The space around is not "used".

IMO should leave it is and the end-user can reconfigure it, very easily.

Reply:
OK, let's leave it. BUT, in the example above the top/bottom padding is lost when header spans multiple lines. Whenever this happens IMO the padding should be kept, so that the entire row is higher.

Reply:
I double-checked how it is in the regular TD in our DataGrid, we use `white-space: nowrap;`. Let's apply it in the header too. 

Mea culpa!

Reply:
Nobody's fault! Users can still override if they want to, just the other way. Consistency is always nice ;-)